topdown/cache/cache.go

```diff
@@ -138,6 +138,9 @@ func (c *cache) unsafeInsert(k ast.Value, v InterQueryCacheValue) (dropped int)
 		}
 	}
 
+	// By deleting the old value, if it exists, we ensure the usage variable stays correct
```

Comment:
Can we also explain the scenario where we need to take this action. For eg, the concurrent request case.

Reply:
This fix simply prevents code like this from breaking the internal state of the cache:
```
	cache.Insert(ast.StringTerm("foo6").Value, cacheValue6)
	cache.Insert(ast.StringTerm("foo6").Value, cacheValue6)
```
I don't see the need to document very explicitly who might call `cache.Insert` in such a way or why they might do it. At least I wouldn't know how to write that documentation. Feel free to update the comment or give me a suggestion on how to write it